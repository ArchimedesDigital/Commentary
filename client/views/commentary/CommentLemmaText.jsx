import FlatButton from 'material-ui/FlatButton';

export default class CommentLemmaText extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			expanded: false,
		};
	}

	_toggleExpanded() {
		const { expanded } = this.state;

		this.setState({
			expanded: !expanded,
		});
	}

	render() {
		const { lines } = this.props;
		const { expanded } = this.state;

		const textLenghtBound = 10;
		const longText = lines.length >= textLenghtBound;

		// if long text
		if (longText) {
			const loops = expanded ? lines.length : textLenghtBound;
			const linesXML = [];

			// construct the XML to be rendered
			// number of lines depended on the expanded state
			for (let i = 0; i < loops; i += 1) {
				linesXML.push(
					<p
						key={i}
						className="lemma-text"
						dangerouslySetInnerHTML={{ __html: lines[i].html }}
					/>);
			}

			return (
				<div>
					{linesXML}
					<div>
						<FlatButton
							label={expanded ? 'Show less' : 'Show more'}
							icon={expanded ? <i className="mdi mdi-chevron-up" /> : <i className="mdi mdi-chevron-down" />}
							onClick={this._toggleExpanded.bind(this)}
						/>
					</div>
				</div>
			);
		}

		// if not lonText
		return (
			<div>
				{lines.map((line, i) => (
					<p
						key={i}
						className="lemma-text"
						dangerouslySetInnerHTML={{ __html: line.html }}
					/>
				))}
			</div>
		);
	}
}

CommentLemmaText.propTypes = {
	lines: React.PropTypes.arrayOf(React.PropTypes.shape({
		html: React.PropTypes.string.isRequired,
		n: React.PropTypes.number.isRequired,
	})).isRequired,
}

