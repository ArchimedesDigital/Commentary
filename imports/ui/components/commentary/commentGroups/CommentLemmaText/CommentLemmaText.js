import React from 'react';
import FlatButton from 'material-ui/FlatButton';

class CommentLemmaText extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			expanded: false,
		};
	}

	_toggleExpanded() {
		const { expanded } = this.state;

		this.setState({
			expanded: !expanded,
		});
	}

	render() {
		const { lines } = this.props;
		const { expanded } = this.state;

		const textLenghtBound = 10;
		const longText = lines.length > textLenghtBound;

		// if long text
		if (longText) {
			const loops = expanded ? lines.length : textLenghtBound;
			const linesXML = [];

			// construct the XML to be rendered
			// number of lines depended on the expanded state
			for (let i = 0; i < loops; i += 1) {
				linesXML.push(
					<div
						key={i}
						className="lemma-text-line"
						style={{display: 'flex', justifyContent: 'flex-start', alignItems: 'baseline'}}
					>
						<span className={`line-n ${(lines[i].n % 5) === 0 ? 'line-n--visible' : ''}`}>
							{lines[i].n}
						</span>
						<p								
							className="lemma-text"
							dangerouslySetInnerHTML={{ __html: lines[i].html }}
						/>
					</div>
				);
			}

			return (
				<div style={{float: 'left'}}>
					{linesXML}
					<div>
						<FlatButton
							label={expanded ? 'Show less' : 'Show more'}
							icon={expanded ? <i className="mdi mdi-chevron-up" /> : <i className="mdi mdi-chevron-down" />}
							onClick={this._toggleExpanded.bind(this)}
						/>
					</div>
				</div>
			);
		}

		// if not longText
		return (
			<div>
				{lines.map((line, i) => (
					<div
						key={i}
						className="lemma-text-line"
						style={{display: 'flex', justifyContent: 'flex-start', alignItems: 'baseline'}}
					>
						<span className={`line-n ${(line.n % 5) === 0 ? 'line-n--visible' : ''}`}>
							{line.n}
						</span>
						<p
							className="lemma-text"
							dangerouslySetInnerHTML={{ __html: line.html }}
						/>
					</div>
				))}
			</div>
		);
	}
}

CommentLemmaText.propTypes = {
	lines: React.PropTypes.arrayOf(React.PropTypes.shape({
		html: React.PropTypes.string.isRequired,
		n: React.PropTypes.number.isRequired,
	})).isRequired,
};

export default CommentLemmaText;
