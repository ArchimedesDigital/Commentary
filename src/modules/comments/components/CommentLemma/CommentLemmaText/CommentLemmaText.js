import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';

class CommentLemmaText extends Component {

	constructor(props) {
		super(props);

		this.state = {
			expanded: false,
		};
	}

	_toggleExpanded() {
		const { expanded } = this.state;

		this.setState({
			expanded: !expanded,
		});
	}

	render() {
		const { textNodes } = this.props;
		const { expanded } = this.state;

		const textLenghtBound = 10;
		const longText = textNodes.length > textLenghtBound;

		// if long text
		if (longText) {
			const loops = expanded ? textNodes.length : textLenghtBound;
			const textNodesHTML = [];

			// construct the HTML to be rendered
			// number of textNodes depended on the expanded state
			for (let i = 0; i < loops; i += 1) {
				textNodesHTML.push(
					<div
						key={i}
						className="lemma-text-line"
					>
						<span
							className={`
								line-n ${(textNodes[i].n % 5) === 0  && textNodes[i].n !== 0 ?
									'line-n--visible'
								:
									''}
							`}
						>
							{textNodes[i].n}
						</span>
						<p
							className="lemma-text"
							dangerouslySetInnerHTML={{ __html: textNodes[i].html }}
						/>
					</div>
				);
			}

			return (
				<div className="comment-lemma-text">
					{textNodesHTML}
					<div
						className="commentary-lemma-show-more-button"
					>
						<FlatButton
							label={expanded ? 'Show less' : 'Show more'}
							icon={expanded ? <i className="mdi mdi-chevron-up" /> : <i className="mdi mdi-chevron-down" />}
							onClick={this._toggleExpanded.bind(this)}
						/>
					</div>
				</div>
			);
		}

		// if not longText
		return (
			<div className="comment-lemma-text">
				{textNodes.map((line, i) => (
					<div
						key={i}
						className="lemma-text-line"
						style={{display: 'flex', justifyContent: 'flex-start', alignItems: 'baseline'}}
					>
						<span className={`line-n ${(line.n % 5) === 0 ? 'line-n--visible' : ''}`}>
							{line.n}
						</span>
						<p
							className="lemma-text"
							dangerouslySetInnerHTML={{ __html: line.html }}
						/>
					</div>
				))}
			</div>
		);
	}
}

CommentLemmaText.propTypes = {
	textNodes: PropTypes.arrayOf(PropTypes.shape({
		html: PropTypes.string.isRequired,
		n: PropTypes.number.isRequired,
	})).isRequired,
};

export default CommentLemmaText;
