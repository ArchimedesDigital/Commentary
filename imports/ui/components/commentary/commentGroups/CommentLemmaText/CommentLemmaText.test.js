import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentLemmaText from './CommentLemmaText';

describe('CommentLemmaText', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentLemmaText lines={[]} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
