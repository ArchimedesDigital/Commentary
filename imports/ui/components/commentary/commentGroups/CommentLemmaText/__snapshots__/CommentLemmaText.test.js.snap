// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentLemmaText renders correctly 1`] = `<div />`;
